CMakeLists.txt

```diff
@@ -0,0 +1,173 @@
+cmake_minimum_required(VERSION 3.25...3.29)
+
+#[=============================================================================[
+#                           Basic project definition                           #
+]=============================================================================]
+
+list(APPEND CMAKE_MESSAGE_CONTEXT Wannier90)
+project(Wannier90
+		VERSION 4.0.0
+		DESCRIPTION "Compute maximally-localised Wannier functions."
+		HOMEPAGE_URL https://www.wannier90.org
+		LANGUAGES Fortran C
+		)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_C_EXTENSIONS OFF)
+
+if (NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif ()
+
+#[=============================================================================[
+#                                   Options                                   #
+]=============================================================================]
+
+option(WANNIER90_MPI "Wannier90: Build with MPI support" OFF)
+option(WANNIER90_SHARED_LIBS "Wannier90: Build library as a shared library" ${PROJECT_IS_TOP_LEVEL})
+option(WANNIER90_INSTALL "Wannier90: Install files" ${PROJECT_IS_TOP_LEVEL})
+option(WANNIER90_TEST "Wannier90: Build test-suite" ${PROJECT_IS_TOP_LEVEL})
+
+#[=============================================================================[
+#                            Project configuration                            #
+]=============================================================================]
+
+if (NOT CMAKE_Fortran_MODULE_DIRECTORY)
+	set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/fortran_mods)
+endif ()
+set(BUILD_SHARED_LIBS ${WANNIER90_SHARED_LIBS})
+
+if (WANNIER90_INSTALL)
+	include(GNUInstallDirs)
+	include(CMakePackageConfigHelpers)
+
+	# CMake does not properly support fortran module installation paths.
+	# https://gitlab.kitware.com/cmake/cmake/-/issues/19608
+	# https://discourse.cmake.org/t/api-design-c-modules-source-listings-and-interface-properties/5389/14
+	# Note: This does not follow fortran-stdlib format, instead it follows Fedora guidelines
+	# https://pagure.io/packaging-committee/issue/1334
+	cmake_path(APPEND CMAKE_INSTALL_LIBDIR "${CMAKE_Fortran_COMPILER_ID}-${CMAKE_Fortran_COMPILER_VERSION}" modules
+			OUTPUT_VARIABLE _DEFAULT_CMAKE_INSTALL_MODULEDIR
+	)
+	set(CMAKE_INSTALL_MODULEDIR ${_DEFAULT_CMAKE_INSTALL_MODULEDIR}
+			CACHE STRING
+			"Fortran module installation path (Not a cmake native variable)"
+	)
+	cmake_path(IS_ABSOLUTE CMAKE_INSTALL_MODULEDIR _is_absolute)
+	if (_is_absolute)
+		set(CMAKE_INSTALL_FULL_MODULEDIR ${CMAKE_INSTALL_MODULEDIR})
+	else ()
```

Comment:
It's not a good idea to use `CMAKE_` prefixed variables (To avoid a possible future conflict).
I think `WANNIER_INSTALL_MODULEDIR` is better.

Reply:
Fully aware of that. The issue is that CMake install support for Fortran has not been implemented and these are a bit tricky because these are ABI dependent and there is no discussion on how it should be designed. This path is also one that should be common among all dependencies, thus the preference of `CMAKE_` variable.

Currently this is made compatible with [`fortran-stdlib`](https://github.com/fortran-lang/stdlib/blob/master/src/CMakeLists.txt) in hopes that there can be convergence towards that.